### PR TITLE
instrument scrapbook using middleware

### PR DESCRIPTION
**Pull request by JosiasAurel**
every request targeted at /api is intercepted by the middleware, which makes the request to the appropriate api function and returns the response.

by doing so it also captures response status codes and sends it to the /api/metric function to send the metric to grafana

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  if (req.url.includes("metric")) return NextResponse.json({ success: true });
+
+  try {
+    let response;
+    if (req.body) {
+      const rawBody = await req.body.getReader().read();
+      const body = JSON.parse(Buffer.from(rawBody?.value).toString("utf8"));
+      // const reqHeaders = Object.fromEntries(req.headers.entries());
+
+      response = await fetch(req.url, {
```
Comment (grymmy) [inline middleware.js:35]: What happens if this fetch takes 10 seconds to resolve.  Will the underlying request that kicked off this middleware API call block for 10s?

File: components/mention.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import Link from 'next/link'
 
 export const StaticMention = memo(
   ({ user = {}, className = '', size = 24, children, ...props }) => (
-    <Link href={`/${user.username}`}>
+    <Link href={`/${user.username}`} legacyBehavior>
```
Comment (grymmy) [inline components/mention.js:8]: Can you ELI5 what this does?

**Review by grymmy — CHANGES_REQUESTED** [review]

File: components/mention.js (hunk 2 above)
Comment (JosiasAurel) [reply components/mention.js:8]: i upgraded the version of next.js and it won't stop showing warning about that Link component 
see [docs](https://nextjs.org/docs/pages/api-reference/components/link#legacybehavior:~:text=legacyBehavior-,An,to%20upgrade.%20A%20codemod%20is%20available%20to%20automatically%20upgrade%20your%20code.,-Good%20to%20know)

File: middleware.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        headers: {
+          "Cookie": cookie
+        }
+      });
+    }
+
+    const data = await response.json();
+
+    // send metric and ignore failure
```
Comment (grymmy) [inline middleware.js:58]: Suggest making this comment a bit more accurate - attempt to send metric, but early-out after 150ms.  Additionally (nitpick!) would suggest making the timeoutController created with a function (createTimeoutPromise?) that takes in the amount of time as a parameter - this would make the line nearly self-documented

**Review by grymmy — APPROVED** [review]
Approved w/ some nitpick comments
